schematicbrushreborn-core/src/main/java/de/eldoria/schematicbrush/config/sections/presets/DatabasePresetRegistry.java

```diff
@@ -0,0 +1,284 @@
+                preparedStatement.setString(1, this.toJson(preset));
+                preparedStatement.execute();
+            } catch (SQLException exception) {
+                this.plugin.getLogger().log(Level.SEVERE, exception.getMessage());
+            }
+        });
+    }
+
+    private void addToDatabase(Player player, Preset preset) {
+        CompletableFuture.runAsync(() -> {
+            try(Connection connection = this.dataSource.getConnection(); PreparedStatement preparedStatement = connection.prepareStatement("INSERT INTO player_presets (PLAYER_UUID, PLAYER_PRESET) VALUES(?, ?)")) {
+                preparedStatement.setString(1, player.getUniqueId().toString());
+                preparedStatement.setString(2, this.toJson(preset));
+                preparedStatement.execute();
+            } catch (SQLException exception) {
+                this.plugin.getLogger().log(Level.SEVERE, exception.getMessage());
+            }
+        });
+    }
+
+    private void removeFromDatabase(String name) {
+        Optional<Preset> preset = this.globalPresets.getPreset(name);
+
+        if(preset.isEmpty()) {
+            return;
+        }
+
+        CompletableFuture.runAsync(() -> {
+            try(Connection connection = this.dataSource.getConnection(); PreparedStatement preparedStatement = connection.prepareStatement("DELETE FROM global_presets WHERE GLOBAL_PRESET=?")) {
+                preparedStatement.setString(1, this.toJson(preset.get()));
+                preparedStatement.execute();
+            } catch (SQLException exception) {
+                this.plugin.getLogger().log(Level.SEVERE, exception.getMessage());
+            }
+        });
+    }
+
+    private void removeFromDatabase(Player player, String name) {
+        Optional<Preset> preset = this.playerPresets.get(player.getUniqueId()).getPreset(name);
+
+        if(preset.isEmpty()) {
+            return;
+        }
+
+        CompletableFuture.runAsync(() -> {
+            try(Connection connection = this.dataSource.getConnection(); PreparedStatement preparedStatement = connection.prepareStatement("DELETE FROM player_presets WHERE PLAYER_UUID=?, PLAYER_PRESET=?")) {
+                preparedStatement.setString(1, player.getUniqueId().toString());
+                preparedStatement.setString(2, this.toJson(preset.get()));
+                preparedStatement.execute();
+            } catch (SQLException exception) {
+                this.plugin.getLogger().log(Level.SEVERE, exception.getMessage());
+            }
+        });
+    }
+
+    private String toJson(Preset preset) {
+        return GSON.toJson(preset);
+    }
+
+    private Preset fromJson(String json) {
+        return GSON.fromJson(json, Preset.class);
+    }
+
+    @Override
+    @NotNull
+    public Map<String, Object> serialize() {
```

Comment:
This seems to be a lot of duplicated code I think.
Maybe moving this to a abstract class in between might have been cleaner. Especially with your current implementation which doesnt access the db for synced calls.